business/jaeger.go

```diff
@@ -0,0 +1,261 @@
+package business
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/kiali/kiali/models"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/log"
+	"github.com/kiali/kiali/util/httputil"
+)
+
+type JaegerService struct {
+	auth          config.Auth
+	businessLayer *Layer
+}
+
+func (in *JaegerService) makeRequest(endpoint string, body io.Reader) (response []byte, status int, err error) {
+	response = nil
+	status = 0
+	client, err := in.getClient()
+	if err != nil {
+		return
+	}
+	req, err := http.NewRequest(http.MethodGet, endpoint, body)
+	if err != nil {
+		return
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return
+	}
+	defer resp.Body.Close()
+	response, err = ioutil.ReadAll(resp.Body)
+	status = resp.StatusCode
+	return
+}
+
+func (in *JaegerService) getClient() (client http.Client, err error) {
+	timeout := time.Duration(1000 * time.Millisecond)
```

Comment:
When I had a lot of traces, this timeout was too easily reached, so I moved to 5 seconds. Not sure what's the best to use ... or maybe we should let the caller specify the timeout, 1s is probably ok the get a single trace, but not to get all traces.